Week3/homework/step2-6.js

```diff
@@ -4,13 +4,27 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
+  const flattenedArr = [];
   // Replace this comment and the next line with your code
-  console.log(arr);
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr[i].length; j++) {
+      flattenedArr.push(arr[i][j]);
+    }
+  }
+  return flattenedArr;
 }
 
 function flattenArray3d(arr) {
+  const flattenedArr = [];
   // Replace this comment and the next line with your code
-  console.log(arr);
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr[i].length; j++) {
+      for (let k = 0; k < arr[i][j].length; k++) {
+        flattenedArr.push(arr[i][j][k]);
+      }
+    }
+  }
+  return flattenedArr;
 }
 
 console.log(flattenArray2d(arr2d)); // -> [1, 2, 3, 4, 5, 6]
```

Comment:
Okay, this works fine for the arrays provided. Can you think of a way to deal with arrays of any dimension?